src/SourceBuild/content/test/Microsoft.DotNet.Installer.Tests/LinuxInstallerTests.cs

```diff
@@ -0,0 +1,365 @@
+
+        InitializeContext(PackageType.Rpm);
+
+        DistroTest($"{repo}:{tag}", PackageType.Rpm);
+    }
+
+    [ConditionalTheory(typeof(LinuxInstallerTests), nameof(IncludeDebTests))]
+    [InlineData(RuntimeDepsRepo, $"{RuntimeDepsVersion}-trixie-slim")]
+    public void DebTest(string repo, string tag)
+    {
+        InitializeContext(PackageType.Deb);
+
+        DistroTest($"{repo}:{tag}", PackageType.Deb);
+    }
+
+    private void InitializeContext(PackageType packageType)
+    {
+        if (packageType == PackageType.Rpm && !_rpmContextInitialized)
+        {
+            // For rpm enumerate RPM packages, excluding those that contain ".cm." in the name
+            List<string> rpmPackages = Directory.GetFiles(Config.AssetsDirectory, "*.rpm", SearchOption.AllDirectories)
+                .Where(p => !Path.GetFileName(p).Contains("-cm.") && !Path.GetFileName(p).EndsWith("azl.rpm"))
+                .ToList();
+
+            foreach (string rpmPackage in rpmPackages)
+            {
+                File.Copy(rpmPackage, Path.Combine(_contextDir, Path.GetFileName(rpmPackage)));
+            }
+
+            DownloadFileAsync(NetStandard21RpmPackage, Path.Combine(_contextDir, Path.GetFileName(NetStandard21RpmPackage))).Wait();
+            _rpmContextInitialized = true;
+        }
+        else if (!_debContextInitialized)
+        {
+            // Copy all DEB packages as well
+            foreach (string debPackage in Directory.GetFiles(Config.AssetsDirectory, "*.deb", SearchOption.AllDirectories))
+            {
+                File.Copy(debPackage, Path.Combine(_contextDir, Path.GetFileName(debPackage)));
+            }
+
+            // Download NetStandard 2.1 packages
+            DownloadFileAsync(NetStandard21DebPackage, Path.Combine(_contextDir, Path.GetFileName(NetStandard21DebPackage))).Wait();
+            _debContextInitialized = true;
+        }
+
+        if (!_sharedContextInitialized)
+        {
+            // Copy nuget packages
+            string nugetPackagesDir = Path.Combine(_contextDir, "packages");
+            Directory.CreateDirectory(nugetPackagesDir);
+            foreach (string package in Directory.GetFiles(Config.PackagesDirectory, "*.nupkg", SearchOption.AllDirectories))
+            {
+                File.Copy(package, Path.Combine(nugetPackagesDir, Path.GetFileName(package)));
+            }
+
+            // Copy and update NuGet.config from scenario-tests repo
+            string newNuGetConfig = Path.Combine(_contextDir, "NuGet.config");
+            File.Copy(Config.ScenarioTestsNuGetConfigPath, newNuGetConfig);
+            InsertLocalPackagesPathToNuGetConfig(newNuGetConfig, "/packages");
+
+            // Find the scenario-tests package and unpack it to the context dir, subfolder "scenario-tests"
+            string? scenarioTestsPackage = Directory.GetFiles(nugetPackagesDir, "Microsoft.DotNet.ScenarioTests.SdkTemplateTests*.nupkg", SearchOption.AllDirectories).FirstOrDefault();
```

Comment:
I'm struggling to follow the execution flow here. Where is that package coming from?

Reply:
Does this project have a dependency on the scenario-tests.proj build?

Reply:
`Microsoft.DotNet.ScenarioTests.SdkTemplateTests*.nupkg` package is a product of the `scenario-tests` repo. I did mention this yesterday at the meeting as a need to have this repo built separately from testing it - as it is today. 

However, the plan is for installer tests to run during `VMR Validation` stage, and download all packages produced during VMR build. Scenario-tests package should be built, if not, that should be changed.

I will be testing the second part of installer-tests work soon, the pipeline hook-up.

Reply:
If the abovementioned plan is not possible, I'd add a dependency on scenario-tests repo project and have it built as part of installer testing.